test/catch2_test_warp_store.cu

```diff
@@ -0,0 +1,296 @@
+            int ITEMS_PER_THREAD,
+            typename T,
+            typename OutputIteratorT>
+  __device__ void operator()(
+      cub::WarpStore<T, ITEMS_PER_THREAD, StoreAlgorithm, LOGICAL_WARP_THREADS> store,
+      OutputIteratorT output,
+      T (&reg)[ITEMS_PER_THREAD]) {
+    store.Store(output, reg);
+  }
+};
+
+template <cub::WarpStoreAlgorithm StoreAlgorithm,
+          int LOGICAL_WARP_THREADS,
+          int ITEMS_PER_THREAD,
+          int TOTAL_WARPS,
+          typename T>
+thrust::device_vector<T>
+compute_reference(int valid_items)
+{
+  const int tile_size = LOGICAL_WARP_THREADS * ITEMS_PER_THREAD;
+  const int total_item_count = TOTAL_WARPS*tile_size;
+  thrust::device_vector<T> d_input(total_item_count);
+
+  if (StoreAlgorithm == cub::WarpStoreAlgorithm::WARP_STORE_STRIPED) {
+    thrust::host_vector<T> input(total_item_count);
+    FillStriped<ITEMS_PER_THREAD, LOGICAL_WARP_THREADS,
+      ITEMS_PER_THREAD * TOTAL_WARPS>(input.begin());
+    d_input = input;
+  } else {
+    c2h::gen(c2h::modulo_t{d_input.size()}, d_input);
+  }
+  if (valid_items != total_item_count) {
+    for (int warp_id = 0; warp_id < TOTAL_WARPS; warp_id++)
+    {
+      thrust::fill(d_input.begin() + warp_id * tile_size + valid_items,
+                   d_input.begin() + (warp_id + 1) * tile_size,
+                   T{});
+    }
+  }
+  return d_input;
+}
+
+// %PARAM% LWT lwt 4:16:32
+// %PARAM% ALGO_TYPE alg 0:1:2:3
+
+using types = c2h::type_list<std::uint8_t, std::uint16_t, std::int32_t, std::int64_t>;
+using items_per_thread = c2h::enum_type_list<int, 1, 4, 7>;
+using logical_warp_threads = c2h::enum_type_list<int, LWT>;
+using algorithms =
+  c2h::enum_type_list<cub::WarpStoreAlgorithm,
+                      cub::WarpStoreAlgorithm::WARP_STORE_DIRECT,
+                      cub::WarpStoreAlgorithm::WARP_STORE_STRIPED,
+                      cub::WarpStoreAlgorithm::WARP_STORE_TRANSPOSE,
+                      cub::WarpStoreAlgorithm::WARP_STORE_VECTORIZE>;
+using algorithm = c2h::enum_type_list<cub::WarpStoreAlgorithm, c2h::get<ALGO_TYPE, algorithms>::value>;
+
+using cache_store_modifier =
+  c2h::enum_type_list<cub::CacheStoreModifier,
+                      cub::CacheStoreModifier::STORE_DEFAULT,
+                      cub::CacheStoreModifier::STORE_WB,
+                      cub::CacheStoreModifier::STORE_CG,
+                      cub::CacheStoreModifier::STORE_CS,
+                      cub::CacheStoreModifier::STORE_WT,
+                      cub::CacheStoreModifier::STORE_VOLATILE>;
+
+constexpr int guarded_store_tests_count = 30;
+
+template <class TestType>
+struct params_t
+{
+  using type = typename c2h::get<0, TestType>;
+
+  static constexpr int logical_warp_threads = c2h::get<1, TestType>::value;
+  static constexpr int items_per_thread = c2h::get<2, TestType>::value;
+  static constexpr cub::WarpStoreAlgorithm algorithm = c2h::get<3, TestType>::value;
+  enum {
+      MAX_WARPS = 2,
```

Comment:
We are using constexpr for the rest of the fields. Is there anything blocking us from doing that here?